geowebcache/swiftblob/src/main/java/org/geowebcache/swift/SwiftUploadTask.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Tobias Schulmann, Catalyst IT Ltd NZ, Copyright 2020
+ */
+package org.geowebcache.swift;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geowebcache.storage.BlobStoreListenerList;
+import org.jclouds.http.HttpResponseException;
+import org.jclouds.io.Payload;
+import org.jclouds.openstack.swift.v1.domain.SwiftObject;
+import org.jclouds.openstack.swift.v1.features.ObjectApi;
+
+public class SwiftUploadTask implements Runnable {
+    static final Log log = LogFactory.getLog(SwiftUploadTask.class);
+    static final String logStr = "%s, %s, %dms";
+
+    private final String key;
+    private final SwiftTile tile;
+    private final ObjectApi objectApi;
+    private final BlobStoreListenerList listeners;
+
+    public SwiftUploadTask(
+            String key, SwiftTile tile, BlobStoreListenerList listeners, ObjectApi objectApi) {
+        this.key = key;
+        this.tile = tile;
+        this.objectApi = objectApi;
+        this.listeners = listeners;
+    }
+
+    private void checkExisted() {
+        if (listeners.isEmpty()) {
+            return;
+        }
+        LocalDateTime time = LocalDateTime.now();
+        long getWithoutBody = System.nanoTime();
+        SwiftObject object = objectApi.getWithoutBody(key);
+        log.info(
```

Comment:
This should really be a debug or trace level rather than info; same goes for the one in the `run` method below; the point is just to get a rough idea of request/response times during development.

The formatting here is doing __some__ work, so it's also one of those occasions where wrapping it in an `if` statement wouldn't hurt.